src/query/anonymization.c

```diff
@@ -188,15 +188,30 @@ static void rewrite_count_histogram(Aggref *aggref, List *aid_refs)
   append_aid_args(aggref, aid_refs);
 }
 
-/* Intended for the denominator in the `avg(col)` rewritten to `sum(col) / count(col)`. */
-static Aggref *make_anon_count_value_aggref(const Aggref *source_aggref)
+/*
+ * Intended for the denominator in the `avg(col)` rewritten to `sum(col) / nullif(count(col), 0)`.
+ * `nullif` is necessary to handle cases where large negative noise brings `count` down to 0
+ * during global aggregation.
+ */
+static Expr *make_safe_anon_count_value(const Aggref *source_aggref)
 {
   Aggref *count_aggref = copyObjectImpl(source_aggref);
   count_aggref->aggfnoid = g_oid_cache.anon_count_value;
   count_aggref->aggtype = INT8OID;
   count_aggref->aggstar = false;
   count_aggref->aggdistinct = false;
-  return count_aggref;
+
+  NullIfExpr *nullif = makeNode(NullIfExpr);
+  nullif->opno = 410; /* = for int8eq. */
```

Comment:
Any define for this? It's very brittle to put such constants which have no guarantee of being backwards compatible.

Reply:
Nothing I could find in pg codebase, there aren't many facilities for operators :(. We can add our own if you prefer, I thought at first that's overkill, but maybe not. There's probably a way to dig it out of the catalogs, I can check how much cruft would that add.